src/coreclr/System.Private.CoreLib/System.Private.CoreLib.csproj

```
@@ -11,16 +11,17 @@
     <!-- Force System.Private.CoreLib.dll into a special IL output directory -->
     <OutputPath>$(RuntimeBinDir)IL\</OutputPath>
     <Configurations>Debug;Release;Checked</Configurations>
-    <Platforms>x64;x86;arm;arm64</Platforms>
+    <Platforms>x64;x86;arm;arm64;wasm</Platforms>
```

`wasm` is not needed.

@jkotas, should we remove this line completely, per https://github.com/dotnet/sdk/pull/48599? (default to AnyCPU)

---

I think this and the previous line are needed to make the project to work inside Visual Studio. It should not be required for command line builds. Visual Studio wants to know all possible configurations that the project support, and it cannot make the list up from a thin air.

---

Ah true, it corresponds to https://github.com/dotnet/runtime/blob/4309e436866720f919d16be0fe534338cd63345e/src/coreclr/System.Private.CoreLib/System.Private.CoreLib.sln#L46-L64

So just delete wasm as it has no meaning in VS context:

```suggestion
    <Platforms>x64;x86;arm;arm64</Platforms>
```

---

Yes, that's what I meant. If we are not adding more archs in SLN, then having them in proj file has no meaning. This is what I got from @akoeplinger in one of my SDK PR.

---

I think about SLNs as auto-generated artifacts from .csproj that is the source of truth. SLN are impossible to write manually. They are checked in just so people do not have to delete them manually every time VS decides to save them.

---

@ViktorHofer can correct me but I think we use slngen tool for this purpose to keep sln files sanitized. If browser-wasm corelib builds from VS, then we could add other targets which  we build on Windows (like android-arm64)?

---

slngen works for libraries. I do not think it works outside of libraries partition.

---

> If browser-wasm corelib builds from VS

FWIW, it is not important that the `.sln` builds, just that it picks up the right ifdefs and files, so that IDE services work (and for that you indeed need the `wasm` config).

It has been a routine workflow downstream to work on the WASM NAOT CoreLib in VS while building it on the command line.

---

> It has been a routine workflow downstream to work on the WASM NAOT CoreLib

Good to know. I just saw mono is shipping wasm for years without it: https://github.com/dotnet/runtime/blob/ca72cdf4fc66ced0c046f3598ceed6646f48989a/src/mono/System.Private.CoreLib/System.Private.CoreLib.csproj#L13

> it is not important that the `.sln` builds

Seeing we have 68K lines of sln in the repo, some may find it important enough to keep things in sync.

---

> @radekdoulik Could you please update the SLN (delete the current one and save a fresh VS generated one) if you want to keep this change?

I updated it with the configuration manager in VS to make the changes small.

---

This is the first project getting wasm in `<Platforms` in runtime repo. One of a kind config is contrary to `keep it consistent` convention used in the repo. The typical way is to open a separate PR updating all projects with `<Platforms` alike, so the intention and purpose is clear.

---

This is the first project getting wasm in `<Platforms` in runtime repo. One of a kind config is contrary to `keep it consistent` convention used in the repo. The typical way is to open a separate PR updating all projects with `<Platforms` alike, so the intention and purpose are clear.

---

This PR is focused on fixing CoreCLR System.Private.CoreLib build for wasm. It is the right thing to add wasm to `Platforms` as part of that so that one can work on CoreCLR's System.Private.CoreLib in Visual Studio. Folks working on CoreCLR use Visual Studio a lot and care about these projects and SLNs working.

The only other project in the repo where wasm would make sense in `Platforms` is https://github.com/dotnet/runtime/blob/main/src/mono/System.Private.CoreLib/System.Private.CoreLib.csproj . This .csproj does not have it, but the associated .sln does have it - twice: https://github.com/dotnet/runtime/blob/main/src/mono/System.Private.CoreLib/System.Private.CoreLib.sln#L55-L56. Both of configurations are broken in VS. For example, trying to build wasm configuration in VS will produce:
```
Build started at 9:27 PM...
1>------ Skipped Build: Project: System.Private.CoreLib (src\System.Private.CoreLib) ------
```
It suggests that folks working on Mono do not use the Mono SLN file, for wasm at least. I agree that it would be nice if .csproj and SLN for Mono's System.Private.CoreLib worked well inside Visual Studio. Fixing that is outside the scope of this PR, it is not a one-line fix.